loader/klvault/vaultloader.go

```diff
@@ -162,11 +164,26 @@ func (vl *Loader) Load(cs konfig.Values) error {
 	for _, secret := range vl.cfg.Secrets {
 		// we fetch our secret
 		var s *vault.Secret
-		s, err = vl.logicalClient.Read(secret.Key)
-		if err != nil {
+		var sData map[string]interface{}
+
+		k := strings.TrimSpace(secret.Key)
+		k = strings.Trim(k, "/")
+		if k == "" {
 			return err
 		}
 
+		p, err := url.Parse(k)
+		if err != nil {
+			return err
+		}
+		s, err = vl.logicalClient.ReadWithData(k, p.Query())
+		if err != nil {
+			return err
+		}
+		sData, ok := s.Data["data"].(map[string]interface{})
```

Comment:
I think you need to check whether `metadata.version` exists on the data object instead of just testing whether this is true. You may have unintended consequences of testing and casting the `data` value on `s.Data`, such as that the user of Vault could have used `data` as a key in their secret in the version 1 kv store.